crates/ursa-index-provider/src/config.rs

```diff
@@ -1,24 +1,34 @@
 use serde::{Deserialize, Serialize};
 use std::path::PathBuf;
 
-const DEFAULT_DB_PATH_STR: &str = "~/.ursa/data/index_provider_db";
-
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
 pub struct ProviderConfig {
-    /// a domain where provider is listening dns/test-node.provider.ursa.earth
+    /// a domain where the node is listening, eg. `dns/test-node.ursa.earth`
+    #[serde(default)]
     pub domain: String,
-    /// indexer url to point to e.g. https://dev.cid.contact
+    /// indexer url to point to, eg. https://dev.cid.contact
+    #[serde(default = "ProviderConfig::default_indexer_url")]
     pub indexer_url: String,
     /// database_path for index provider db
+    #[serde(default = "ProviderConfig::default_database_path")]
     pub database_path: PathBuf,
 }
 
+impl ProviderConfig {
+    fn default_database_path() -> PathBuf {
```

Comment:
Like why these functions returning a single value? Surely instantiating a map or something else would be more pragmatic and still serve the dynamic config feature

Reply:
Serde derive supports functions only per field defaults

```
#[serde(default = "function")]
fieldname: String
```

Using `#[serde(default)]` alone results in Default::default() being called for the type specifically not the struct field containing the type